src/layout.jl

```diff
@@ -102,11 +102,11 @@ julia> locs_x, locs_y = spring_layout(g)
 ```
 """
 function spring_layout(g::AbstractGraph,
-                       locs_x=2*rand(nv(g)).-1.0,
-                       locs_y=2*rand(nv(g)).-1.0;
+                       locs_x_in::Vector{R1}=2*rand(nv(g)).-1.0,
+                       locs_y_in::Vector{R2}=2*rand(nv(g)).-1.0;
```

Comment:
Should we really restrict ourselves to `Vector` here?

Reply:
I used Vector here to make it consistent with the same `args` in `gplot()` (line 119 of plots.jl). Do you think we should relax it here?